src/cc/api/BPFTable.h

```diff
@@ -114,14 +123,14 @@ class BPFArrayTable : public BPFTableBase<int, ValueType> {
       throw std::invalid_argument("Table '" + desc.name + "' is not an array table");
   }
 
-  StatusTuple get_value(const int& index, ValueType& value) {
-    if (!this->lookup(const_cast<int*>(&index), &value))
+  virtual StatusTuple get_value(const int& index, ValueType& value) {
```

Comment:
@mauriciovasquezbernal Sorry I missed this during review, but why do we need `virtual` for these methods? We will be having the table instances in correct class types (like `BPFPercpuArrayTable`) and we don't need virtual methods? Did I miss something?

Reply:
Those methods are called from the `operator[]` and `get_table_offline` methods on the base classes (`BPFArrayTable` and `BPFHashTable`)

For instance: https://github.com/iovisor/bcc/blob/master/src/cc/api/BPFTable.h#L140

Reply:
Makes sense. Thanks for the explanation!